apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+      { review_decision: 'approve' as EnumPullReqReviewDecision },
+      { review_decision: 'request_changes' as EnumPullReqReviewDecision }
+    ]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'approve' as EnumPullReqReviewDecision }]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'reviewed' as EnumPullReqReviewDecision }]
+  }
+]
+// Mock data for extractInfoFromRuleViolationArr
+const mockRuleViolations = [
+  {
+    violations: [{ message: 'Violation 1' }, { message: 'Violation 2' }]
+  },
+  {
+    violations: [{ message: 'Violation 3' }]
+  }
+]
+
+// Tests
+describe('processReviewDecision', () => {
+  it('should mark review as outdated if SHAs differ', () => {
+    const result = processReviewDecision(mockReviewDecision, mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(PullReqReviewDecision.outdated)
+  })
+})
+
+describe('generateStatusSummary', () => {
+  it('should generate a summary of status checks', () => {
+    const result = generateStatusSummary(mockChecks)
+    expect(result.summary.total).toBe(2)
+    expect(result.summary.successReq).toBe(1)
+    expect(result.summary.failed).toBe(1)
+  })
+})
+
+// Mock data for determineStatusMessage
+const mockChecksForStatusMessage = [
+  { check: { status: 'success' as EnumCheckStatus }, required: true },
+  { check: { status: 'failure' as EnumCheckStatus }, required: false }
+]
+
+describe('determineStatusMessage', () => {
+  it('should determine the status message based on the status summary', () => {
+    const result = determineStatusMessage(mockChecksForStatusMessage as TypeCheckData[])
+    expect(result).toEqual({
+      color: 'text-success',
+      content: '1 failed, 1 succeeded',
+      status: 'success',
+      title: 'All required checks passed'
+    })
+  })
+})
+
+describe('extractSpecificViolations', () => {
+  it('should extract specific violations based on rule', () => {
+    const result = extractSpecificViolations(mockViolationsData, 'rule1')
+    expect(result).toEqual([{ code: 'rule1', description: 'Violation 1' }])
+  })
+})
+
+describe('checkIfOutdatedSha', () => {
+  it('should return true if SHAs differ', () => {
+    const result = checkIfOutdatedSha(mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(true)
+  })
+})
+
+describe('capitalizeFirstLetter', () => {
+  it('should capitalize the first letter of a string', () => {
+    const result = capitalizeFirstLetter('hello')
+    expect(result).toBe('Hello')
+  })
+})
+
+describe('findChangeReqDecisions', () => {
+  it('should find change request decisions from entries', () => {
+    const result = findChangeReqDecisions(mockEntriesForChangeReq, 'approve')
+    expect(result).toEqual([
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      },
+      {
+        owner_evaluations: [{ review_decision: 'approve' }]
+      }
+    ])
+  })
+})
+
+describe('findWaitingDecisions', () => {
+  it('should find waiting decisions from entries', () => {
```

Comment:
This test doesn't really prove that it can find waiting decisions, rather it just returns an empty array. Can we modify the test data to include waiting decisions that can be picked up?

Reply:
fixed